src/stores/ActiveWidgetStore.js

```diff
@@ -75,8 +75,6 @@ class ActiveWidgetStore extends EventEmitter {
     destroyPersistentWidget() {
         const toDeleteId = this._persistentWidgetId;
 
-        const PersistedElement = sdk.getComponent("elements.PersistedElement");
-        PersistedElement.destroyElement('widget_' + toDeleteId);
```

Comment:
How come this didn't work? Seemed like a better place to do it...

Reply:
'cos it was trying to destroy the one current persistent widget, but in the case of a jitsi widget which got mounted because you were in its room but never became persistent, it wouldn't destroy anything. The subtle difference is tearing down the current persistent widget vs tearing down the widget in the current view which has been mounted in a persisted element.

Reply:
Got it :+1: